lib/internal/test_runner/test.js

```diff
@@ -163,9 +164,16 @@ class Test extends AsyncResource {
   constructor(options) {
     super('Test');
 
-    let { fn, name, parent, skip } = options;
+    let { fn, name, parent, skip, testNamePatterns: _testNamePatterns } = options;
     const { concurrency, only, timeout, todo, signal } = options;
 
+    if (_testNamePatterns && testNamePatterns) {
```

Comment:
is it ok that this allows any falsy value to still be passed? alternatively:
```suggestion
    if (Object.hasOwn(options, '_testNamePatterns') && testNamePatterns) {
```